### DISCUSSION TITLE
New Defence Implementation

### DISCUSSION THREAD

**Discussion by RoeyBokobza** [Trusted-AI/adversarial-robustness-toolbox#2409] [Q&A]
Hello everyone, 

Help would be much appreciated here. I developed a pre-processor as part of my study, which I believe will be helpful to all users of this library. 
How do I push contribute this library's code? 

Thank you!

**Comment by beat-buesser** [comment]
Hi @RoeyBokobza Thank you for considering ART. Could you please provide more details on the implemented algorithm or maybe a link to a paper?

**Reply by RoeyBokobza** [reply]
Thank you for answering. 
The algorithm is presented in this paper : https://arxiv.org/abs/2106.04938. I implemented it, and it worked well for me so I wanted to share this implementation with all ART users.

**Reply by beat-buesser** [reply]
Hi @RoeyBokobza Does the defence add small amounts of random perturbations to the inputs? You could suggest it as a feature by opening a feature request issue and then follow-up with a pull request.
